pkg/agents/team2/policies.go

```diff
@@ -41,3 +55,47 @@ func (a *CustomAgent2) adjustPolicies() {
 		}
 	}
 }
+
+func (a *CustomAgent2) winOrLose(state int) float64 {
+	sumPolicies := 0.0
+	sumAveragePolicies := 0.0
+	Delta_win := 0.025
+	Delta_lose := 0.1
+	for i := 0; i < len(a.actionSpace); i++ {
+		sumPolicies += a.policies[state][i] * a.qTable[state][i]
+		sumAveragePolicies += a.averagePolicies[state][i] * a.qTable[state][i]
+	}
+	if sumPolicies > sumAveragePolicies {
+		return Delta_win
+	}
+	return Delta_lose
+}
+
+func (a *CustomAgent2) exportPolicies() {
+	f, err := os.OpenFile(fmt.Sprintf("%s%s%s", a.ID(), "policies", ".csv"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		log.Println("error opening csv: ", err)
+	}
+	defer f.Close()
+
+	w := csv.NewWriter(f)
+	policies := a.policies
+	// make string table
+	sPolicies := make([][]string, len(policies))
+	for i := 0; i < len(policies); i++ {
+		sPolicies[i] = make([]string, len(policies[0]))
+	}
+
+	// convert float64 policies to string policies
+	for i := 0; i < len(policies); i++ {
+		for j := 0; j < len(policies[0]); j++ {
+			sPolicies[i][j] = fmt.Sprint(policies[i][j])
+		}
+	}
+
+	err = w.WriteAll(sPolicies)
+
+	if err != nil {
+		log.Println("error writing csv:", err)
```

Comment:
Ideally use `log.Error` I think. Will defer to @JaafarRammal  over this though

Reply:
yep